src/Core/Domain/Product/Command/UpdateProductOptionsCommand.php

```diff
@@ -0,0 +1,337 @@
+
+    /**
+     * @var bool|null
+     */
+    private $availableForOrder;
+
+    /**
+     * @var bool|null
+     */
+    private $onlineOnly;
+
+    /**
+     * @var bool|null
+     */
+    private $showPrice;
+
+    /**
+     * @var array|null key value pairs where key is language id and value is the array of tags
+     */
+    private $localizedTags;
+
+    /**
+     * @var ProductCondition|null
+     */
+    private $condition;
+
+    /**
+     * @var Isbn|null
+     */
+    private $isbn;
+
+    /**
+     * @var Upc|null
+     */
+    private $upc;
+
+    /**
+     * @var Ean13|null
+     */
+    private $ean13;
+
+    /**
+     * @var Mpn|null
+     */
+    private $mpn;
+
+    /**
+     * @var Reference|null
+     */
+    private $reference;
+
+    /**
+     * @param int $productId
+     */
+    public function __construct(int $productId)
+    {
+        $this->productId = new ProductId($productId);
+    }
+
+    /**
+     * @return ProductId
+     */
+    public function getProductId(): ProductId
+    {
+        return $this->productId;
+    }
+
+    /**
+     * @return ProductVisibility|null
+     */
+    public function getVisibility(): ?ProductVisibility
+    {
+        return $this->visibility;
+    }
+
+    /**
+     * @param ProductVisibility $visibility
+     *
+     * @return UpdateProductOptionsCommand
+     */
+    public function setVisibility(string $visibility): UpdateProductOptionsCommand
+    {
+        $this->visibility = new ProductVisibility($visibility);
+
+        return $this;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function isAvailableForOrder(): ?bool
+    {
+        return $this->availableForOrder;
+    }
+
+    /**
+     * @param bool $availableForOrder
+     *
+     * @return UpdateProductOptionsCommand
+     */
+    public function setAvailableForOrder(bool $availableForOrder): UpdateProductOptionsCommand
+    {
+        $this->availableForOrder = $availableForOrder;
+
+        return $this;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function isOnlineOnly(): ?bool
+    {
+        return $this->onlineOnly;
+    }
+
+    /**
+     * @param bool $onlineOnly
+     *
+     * @return UpdateProductOptionsCommand
+     */
+    public function setOnlineOnly(bool $onlineOnly): UpdateProductOptionsCommand
+    {
+        $this->onlineOnly = $onlineOnly;
+
+        return $this;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function toShowPrice(): ?bool
+    {
+        return $this->showPrice;
+    }
+
+    /**
+     * @param bool $showPrice
+     *
+     * @return UpdateProductOptionsCommand
+     */
+    public function setShowPrice(bool $showPrice): UpdateProductOptionsCommand
+    {
+        $this->showPrice = $showPrice;
+
+        return $this;
+    }
+
+    /**
+     * @return string[]|null
+     */
+    public function getLocalizedTags(): ?array
+    {
+        return $this->localizedTags;
+    }
+
+    /**
+     * @param string[] $localizedTags
```

Comment:
We should add a comment about the expected format `array of tags in an array indexed by language ID` And maybe add some validation to assert it's a two dimensional array And not just an array of strings unlike other localized fields

Reply:
Unless we make it simpler and use an array of string and the handler is the one that parses the comma string?

Reply:
for now I added a comment, if we see in future that its really worth it, we can change it

Reply:
> And maybe add some validation to assert it's a two dimensional array And not just an array of strings unlike other localized fields

Hey, good idea ! I always wanted some custom assertions for it.
There is a standard way to describe it: <key, value>

So here it would be <key, <string, string>>